lib/parser/index.js

```diff
@@ -0,0 +1,71 @@
+const sketch2json = require('sketch2json');
+const {getClient, figma2json, findIds, flattenChildren} = require('../figma');
+const {
+    getPageArrays, getColorsFromArtboard, getGradientsFromArtboard, getShadowsFromArtboard,
+    getBordersFromArtboard
+  } = require('../../lib/sketch');
+
+const getFileContents = require('../reader');
+
+const getFileExtension = (file) => file.split('.').pop();
+exports.getFileExtension = getFileExtension;
+
+module.exports = async (args, flags) => {
+  const file = args[0];
+  const regex = /.*sketch/;
+  if(regex.test(file)){
+      // file is sketchfile
+      try {
+        const response = await sketch2json(await getFileContents(file));
+        const primitivesPage = getPageArrays(response).find(i => i.name.toLowerCase() === 'primitives');
+        if (!primitivesPage) {
+          throw new Error(`No primitives page found.`)
+        }
+
+        module.getColors = () => flags.useColorArtboards
+          ?  getColorsFromArtboard(primitivesPage.layers) 
+          :  response.document.assets.colorAssets
+              .map(({ color, _class, name }) => ({ ...color, _class, name }));
+
+        module.getGradients = () => flags.useGradientArtboards
+            ? getGradientsFromArtboard(primitivesPage.layers)
+            : response.document.assets.gradientAssets
+                .map(({ gradient, _class }) => ({ ...gradient, _class }));
+        
+        module.getTextStyles = () => response.document.layerTextStyles;
+        module.getShadows = () => getShadowsFromArtboard(primitivesPage.layers);
+        module.getBorders = () => getBordersFromArtboard(primitivesPage.layers);
+        module.getFonts = () => response.meta.fonts;
+        module.getVersion = () => ({sketchVersion: response.meta.appVersion});
+
+        return ({parser: module, response});
+      } catch(err){
+        throw new Error(err);
+      }
+  }
+  // file is figmafile
+  const [fileId, token] = args;
```

Comment:
Token should be supplied as a CLI flag and not an argument. Please take a look at `cli.js` and add a `--token` flag to use. Before parsing anything Figma-related, check if you have that CLI flag, otherwise throw an error.